lib/AVL-tree/AVL-tree.c

```diff
@@ -0,0 +1,308 @@
+#include "AVL-tree.h"
+#include "errorCodes.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+struct Node
+{
+    char* key;
+    char* value;
+    Node* rightChild;
+    Node* leftChild;
+    int height;
+};
+
+struct AVLTree
+{
+    Node* root;
+};
+
+AVLTree* createAVLTree()
+{
+    AVLTree* tree = calloc(1, sizeof(AVLTree));
+    if (tree == NULL)
+    {
+        return NULL;
+    }
+    tree->root = NULL;
+    return tree;
```

Comment:
```suggestion
    return calloc(1, sizeof(AVLTree));
```
Это то же самое :) calloc и так проинициализирует всё нулями, то есть в tree->root будет NULL, а раз так, нам не надо пытаться разыменовать tree, так что мы можем смело его вернуть, будь там NULL или не NULL